_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,419 @@
+
+After looking around, stop minikube before the cloud deployment:
+
+```bash
+minikube stop
+```
+
+## Deploy to DigitalOcean Kubernetes
+
+Now that the architecture works locally, let's proceed to the cloud deployment. First, create a [DigitalOcean](https://cloud.digitalocean.com/registrations/new) account, you can try their services with a free $100 credit that is available for 60 days.
+
+Most of the cluster tasks, if not all, can be accomplished using [doctl](https://github.com/digitalocean/doctl#installing-doctl), the command-line interface (CLI) for the DigitalOcean API. Install the tool, and perform the authentication with DigitalOcean:
+
+```bash
+doctl auth init
+```
+You will be prompted to enter the DigitalOcean access token that you can generate in the DigitalOcean control panel. Sign in, and then in the left menu go to **API**, click **Generate New Token**. Enter a token name, and click **Generate Token**. Copy the new token from the **Tokens/Keys** table.
+
+{% img blog/jhipster-digital-ocean/do-control-panel-token.png alt:"DigitalOcean Control Panel for Token Create" width:"500" %}{: .center-image }
+
+You can find a detailed list of pricing for cluster resources at [DigitalOcean](https://docs.digitalocean.com/products/kubernetes/),  and with `doctl` you can quickly retrieve a list of node size options available for your account:
+
+```bash
+doctl k options sizes
+```
+
+**NOTE**: I tested the cluster with the higher size Intel nodes available for my account, `s-2vcpu-4gb-intel`, in an attempt to run the application in the default cluster configuration, a three-node cluster with a single node pool in the nyc1 region, using the latest Kubernetes version. As I started to see pods not running due to "insufficient CPU", I increased the number of nodes later. DigitalOcean's latest and default Kubernetes version at the moment of writing this post is 1.22.8.
+
+Create the cluster with the following command line:
+
+```bash
+doctl k cluster create do1 -v --size s-2vcpu-4gb-intel
+```
+After creating a cluster, `doctl` adds a configuration context to kubectl and makes it active, so you can start monitoring your cluster right away with k9s. First, apply the resources configuration to the DigitalOcean cluster:
+
+```bash
+./kubectl-apply.sh -f
+```
+
+Monitor the deployment with k9s:
+
+```bash
+k9s -n demo
+```
+
+{% img blog/jhipster-digital-ocean/k9s-do-cluster.png alt:"k9s user interface monitoring DigitalOcean Kubernetes" width:"800" %}{: .center-image }
+
+Once you see the `jhipster-registry` pods are up, set up port forwarding again so you can also monitor the status of the services in the registry UI.
+
+### Increase CPU
+
+The three-node cluster might not provide enough CPU for this microservices architecture, so you might see `insufficient CPU` logs for some pods. You can get the pod events with `kubectl describe`:
+
+```bash
+kubectl describe pod jhipster-registry-0 -n demo
```

Comment:
I tried the following command and didn't see any issues.

```
kubectl describe pod jhipster-registry-0 -n demo | grep cpu
```

Reply:
It also seems kinda strange that there are two instances of the JHipster Registry created. I believe @moksamedia ran into this as well. Maybe it's JHipster's default?

Reply:
In the jhipster-registry.yml I see "replicas: 2", it seems the default when using the k8s generator. The `kubectl describe` line above was just an example, during my test, other pods failed with "insufficient cpu", it can be the registry or other service.

Reply:
OK. Let's leave it as two then.